### DISCUSSION TITLE
Scene override for each blend file in Loom Batch

### DISCUSSION THREAD

**Discussion by ludnny** [p2or/blender-loom#80] [Ideas]
Thank you for this super useful add-on.

In Loom Batch Render, it is possible to add multiple Blender projects. 
But is there any way to choose the Scene to render?
If not, I'd like to request this feature!

**Comment by p2or** [comment]
Hi @ludnny,

thanks for your feature request and your kind words - glad that the add-on is also useful to others.

> In Loom Batch Render ... is there any way to choose the Scene to render?

As of the current state, we can only override the render settings in the batch dialog, see the [presets](https://github.com/p2or/blender-loom?tab=readme-ov-file#render-presets) option. Otherwise the blend file gets rendered as is. As a workaround, the batch dialog also provides a 'snapshot feature' that allows to create a (temporary) snapshot of the current blend file which automatically gets added to the queue:

<img width="762" alt="Screenshot 2024-03-12 at 09 40 59" src="https://github.com/p2or/blender-loom/assets/512368/8cb9e274-fbf8-4109-bcb1-d606db49c96a">
<br><br>

> If not, I'd like to request this feature

Sure, if several people are in favor of this and it really means an improvement for their workflow, I'm all for it. However, I personally don't see the real benefit of such an option right now, since you can utilize a 'file output' node in the compositor to save each layer separately, which also means having full control in a single file and a comprehensible logic of what is/was rendered for the project. Could you please elaborate why you think this would be useful and worth to implement? 

The other issue I see is: Where to put the layer selection within each row without bloating the UI and destroying the current logic...? If you have a good idea or mockup to solve this, please let me know.

Cheers,
Christian

**Comment by ludnny** [comment]
Thank you very much @p2or!

If I understand correctly, I can use the File Output node in the compositer to save each ViewLayer of a Scene separately. 
But what about the Scenes themselves?
A Blender project can have many Scenes, with different Resolution, Frame Start/End etc...  
Is there a way right now to batch render all of them using the AddSnapshot workaround, the preset or the Compositor workaround? 
I didn't find how.

If I take an example:

FileHouse.blend
   SceneLivingroom (1920x1080, 100 frames)
   SceneBedroom (1080x1080, 50 frames)
   
FileCity.blend
   SceneBerlin (1920x1080, 150 frames)
   SceneParis (1080x1080, 75 frames)
   
To be able to batch render the 4 Scenes from 2 Blender Files, I was expecting to find a Scene button in Loom on each line. 
![image](https://github.com/p2or/blender-loom/assets/46849877/f3a252a8-b124-434c-99a3-087bc652644f)

When pressed, it would add a new Scene line (the same way pressing the encode_flag button adds Colorspace Framerate and Codec), with a Scene picker. By default, the default Scene is displayed.

Does it make sens, or am I overthinking this?

Cheers,

**Comment by p2or** [comment]
Hi @ludnny,

> If I understand correctly, I can use the File Output node in the compositer to save each ViewLayer of a Scene separately. But what about the Scenes themselves?

You can render each scene in the same way as you would render the view layers, see [this blend](https://github.com/p2or/blender-loom/files/14571626/l.blend.zip) for reference. The actual comp in the 'Main' scene allows to render all other scenes you want: 

<img width="1420" alt="Screenshot 2024-03-12 at 07 51 22" src="https://github.com/p2or/blender-loom/assets/512368/b8d17112-e264-46c9-a28a-40fd02718ca2">
<br><br>

> Is there a way right now to batch render all of them using the AddSnapshot workaround ?

Yeah, you can open up the blend, make the scene you want to render the 'active' one, hit the snapshot button (which stores the blend into a new folder called `temp`), repeat that for the scenes you'd like to render and add all the generated blend files to the queue - depending on your setup, I guess this will only take a few seconds.

Does this help?

Cheers,
Christian

**Comment by ludnny** [comment]
That helps, thanks!

The Add Snapshot workflow does it for me, feel free to close the issue if there are no other requests.

Cheers,

**Comment by p2or** [comment]
Hi @ludnny,

had some time and implemented that the scene, the camera and the render preset can be overridden per item in the queue. Still not satisfied with the UI though but if you have time, I would be pleased if you could try it out and report back:

https://github.com/p2or/blender-loom/assets/512368/2439d0ef-7901-4039-bc7c-b5a4837affe9

If you are interessted, you can download the [latest changes from the corresponding branch](https://github.com/p2or/blender-loom/tree/batch-overrides-ui) and install *Loom* as usual.

Cheers,
Christian

**Comment by ludnny** [comment]
Hi @p2or, 

That's  amazing, thank you!

I find it quite helpful and intuitive like this (tested successfully on Blender 4.0 on Windows).

Just one thing that could be more intuitive: after choosing a Scene in the batch render window, the Camera picker shows all cameras from the entire project. 
That would be easier to see only the cameras from the selected Scene, or to display the cameras from other scenes greyed out. 
I guess that's not a trivial thing to do, and it's a detail. 
In any case, congrats, that's an amazing improvement!

**Comment by p2or** [comment]
Hi @ludnny,

many thanks for testing and your feedback! 

> ... after choosing a Scene in the batch render window, the Camera picker shows all cameras from the entire project.

An indication that you have really looked into it, thank you very much for your time! Well, as you suspect, it is indeed very expensive to filter the cameras per scene, and would mean a lot of extra code. I promise that as soon as I find a good option, I will add it.

Cheers,
Christian
